### DISCUSSION TITLE
AdGuardHome Go binary size

### DISCUSSION THREAD

**Discussion by jamesmacwhite** [AdguardTeam/AdGuardHome#4024] [General]
First of all, I just want to say AdGuard Home is great and it's really exciting to see all the constant development and features constantly being rolled out.

Once thing I've noticed is the binary size seems to be growing a fair bit. Comparing the last two latest stable releases 0.106.3 to 0.107.0 for ARMv7.

* 0.106.3 - 18.4 MB
* 0.107.0 - 32.1 MB

Roughly a 57% increase. Now for most contexts and platforms, this isn't really an issue. An increase of 13.7 MB in the context of today's world with data centres with literally petabytes of storage, it's nothing! However, for more embedded platforms such as routers which will have limited flash space, this is quite significant. There are of course ways to extend flash storage space with additional storage partitions and such, but it did make me think about the binary size itself.

With the great development cycle and constant releases and improvements is the binary size just going to continue to grow? I'm not really that familiar with Go so please forgive my ignorance. It is my understanding that because the binary is statically linked and includes the Go runtime, it will be much heavier, but it does raise the question whether at some point the binary size will make running on embedded products like routers very difficult.

Either way, just thought it was an interesting topic, thanks for anyone's contributions!

**Comment by ainar-g** [comment]
@jamesmacwhite, hello and thank you for the concerns!  There were several differences in how we package stuff in v0.107.0.  I've filled #4025 to investigate some of that size growth.

**Comment by jamesmacwhite** [comment]
@ainar-g Thank you for raising it as an issue.

Just out of interest I used upx on the 0.107.0 ARMv7 binary and managed to get a compression ratio of about 68%, which made it around 10 MB. Compression will potentially have performance trade offs, but it did make we wonder about compression optimisations generally.
